### DISCUSSION TITLE
Kubernetes pulling image Error: ErrImagePull from private docker registry on nexus

### DISCUSSION THREAD

**Discussion by hesyar** [mysticaltech/terraform-hcloud-kube-hetzner#435] [Q&A] [answered]
Hello there,

I'm not sure whether this is the right place to ask this question but it could be..
So I successfully setup my cluster with the kube.tf file and everything worked so far great. 

I've now also setup nexus based on a config file I used in a previous cluster. Nexus works and I configured to have it a docker registry aswell. I can push my images to the registry. So far so good. Now the problem comes when I deploy an application and point the image to my private registry.

It is not able to fetch the image although I've set the docker secret properly. I'm getting the following error message:

`failed to do request: Head "https://my-image-path/manifests/latest": dial tcp loadbalancerip:443: i/o timeout`

The reason why I post this question here is that I assume it has something to do with the loadBalancer but I have to say that its just that I'm suspected about this since I have set everything like it should. Furthermore when I do a HEAD request to https://my-image-path/manifests/latest on my local machine I get the correct response.

I'm a noob it networking stuff so would be really glad if someone could help me.

Thanks for supporting in advance!

**Comment by mysticaltech** [comment, marked as answer]
You are using Ingress Nginx right? If so, you have to configure the hostname attribute for it to work locally. We have previous issues dealing with that. See https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/issues/409#issuecomment-1312780493

Or just use Traefik for peace of mind.
